lexicon/providers/aurora.py

```diff
@@ -43,8 +43,8 @@ def authenticate(self):
     # Create record. If record already exists with the same content, do nothing'
     def create_record(self, type, name, content):
         data = {'type': type, 'name': self._relative_name(name), 'content': content}
-        if self.options.get('ttl'):
-            data['ttl'] = self.options.get('ttl')
+        if self._get_lexicon_option('ttl'):
```

Comment:
will this retrieve from `lexicon:{{provider}}:ttl` if specified?

Reply:
Not exactly. There is two convenient method: `_get_lexicon_option` and `_get_provider_option`. The two map their call to `self.config.resolve`, the first on `lexicon:key`, the second to `lexicon:provider:option`. Here, ttl is an option for Lexicon, so it will be the key `lexicon:ttl`.

Two ideas here. First, to incite people to use a unified way to access options from a provider, while hiding the internal implementation and simplify development. Second, to make explicit what is relevant to lexicon, and what is for the current provider.

Reply:
But now, I can add a fallback system, certainly in `_get_lexicon_option`, to try to resolve `lexicon:key` with `lexicon:provider:key` if the first one returned None. 

However, it will make integration tests more complicated, because by default, any option not found in the real config sources will be resolved as `placeholder_something`. So testing `lexicon:provider:key` will never occur during an integration test.

If you want that, I think that by default, the fallback lambda should generate placeholders only for parameters relevant to the current provider.

Reply:
nope, I was just curious if I missed some additional logic somewhere in the `ConfigResolver`.
Something to keep in mind for the `auto` provider, some providers have very restrictive ranges for TTL.

Reply:
In the future, we could allow ttl to be overriden for a specific provider.